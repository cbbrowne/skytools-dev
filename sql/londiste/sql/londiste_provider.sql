
set client_min_messages = 'warning';
\set VERBOSITY 'terse'

--
-- tables
--
create table testdata (
    id serial primary key,
    data text
);
create table testdata_nopk (
    id serial,
    data text
);

select current_database();

select * from pgq_node.register_location('aset', 'rnode', 'dbname=db', false);
select * from pgq_node.create_node('aset', 'root', 'rnode', 'londiste_root', null::text, null::int8, null::text);

select * from londiste.local_add_table('aset', 'public.testdata_nopk');
select * from londiste.local_add_table('aset', 'public.testdata');
insert into testdata (data) values ('test-data');
select * from londiste.get_table_list('aset');
select * from londiste.local_show_missing('aset');
select * from londiste.local_remove_table('aset', 'public.testdata');
select * from londiste.local_remove_table('aset', 'public.testdata');
select * from londiste.get_table_list('aset');

select ev_id, ev_type, ev_data, ev_extra1 from pgq.event_template;

select * from londiste.local_show_missing('aset');

