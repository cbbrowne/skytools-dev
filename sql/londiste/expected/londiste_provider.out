set client_min_messages = 'warning';
\set VERBOSITY 'terse'
--
-- tables
--
create table testdata (
    id serial primary key,
    data text
);
create table testdata_nopk (
    id serial,
    data text
);
select current_database();
 current_database 
------------------
 regression
(1 row)

select * from pgq_node.register_location('aset', 'rnode', 'dbname=db', false);
 ret_code |      ret_note       
----------+---------------------
      200 | Location registered
(1 row)

select * from pgq_node.create_node('aset', 'root', 'rnode', 'londiste_root', null::text, null::int8, null::text);
 ret_code |                          ret_note                          
----------+------------------------------------------------------------
      200 | Node "rnode" initialized for queue "aset" with type "root"
(1 row)

select * from londiste.local_add_table('aset', 'public.testdata_nopk');
 ret_code |                      ret_note                      
----------+----------------------------------------------------
      400 | Primary key missing on table: public.testdata_nopk
(1 row)

select * from londiste.local_add_table('aset', 'public.testdata');
 ret_code |           ret_note           
----------+------------------------------
      200 | Table added: public.testdata
(1 row)

insert into testdata (data) values ('test-data');
select * from londiste.get_table_list('aset');
   table_name    | local | merge_state | custom_snapshot | table_attrs | dropped_ddl | copy_role 
-----------------+-------+-------------+-----------------+-------------+-------------+-----------
 public.testdata | t     | ok          |                 |             |             | 
(1 row)

select * from londiste.local_show_missing('aset');
 obj_kind |          obj_name           
----------+-----------------------------
 S        | public.testdata_id_seq
 S        | public.testdata_nopk_id_seq
 r        | public.testdata_nopk
(3 rows)

select * from londiste.local_remove_table('aset', 'public.testdata');
 ret_code |            ret_note            
----------+--------------------------------
      200 | Table removed: public.testdata
(1 row)

select * from londiste.local_remove_table('aset', 'public.testdata');
 ret_code |             ret_note             
----------+----------------------------------
      400 | Table not found: public.testdata
(1 row)

select * from londiste.get_table_list('aset');
 table_name | local | merge_state | custom_snapshot | table_attrs | dropped_ddl | copy_role 
------------+-------+-------------+-----------------+-------------+-------------+-----------
(0 rows)

select ev_id, ev_type, ev_data, ev_extra1 from pgq.event_template;
 ev_id |        ev_type        |              ev_data               |    ev_extra1    
-------+-----------------------+------------------------------------+-----------------
     1 | londiste.add-table    | public.testdata                    | 
     2 | I                     | (id,data) values ('1','test-data') | public.testdata
     3 | londiste.remove-table | public.testdata                    | 
(3 rows)

select * from londiste.local_show_missing('aset');
 obj_kind |          obj_name           
----------+-----------------------------
 S        | public.testdata_id_seq
 S        | public.testdata_nopk_id_seq
 r        | public.testdata
 r        | public.testdata_nopk
(4 rows)

